### ISSUE TITLE
Crafting Glitch?

### ISSUE THREAD

**Issue by hannes213** [SmartlyDressedGames/Unturned-3.x-Community#3844]
I was in the Russia map when I wanted to craft a pump jack. Not noticing that I didn't have all the required materials in my inventory, but rather in crates beside me, I tried to craft it (the crafting place showed that I could craft it. Like the amount of things I actually had in my inventory were showed as doubled there), but when I looked into my inventory and crates and on the ground I couldn't find it anywhere and half of the materials I needed to craft it were completely gone (the metal sheets and bars). The same thing happened when i tried to craft the industrial gas can right after to test and then my gas cans were gone.

**Comment by SDGNelson** [comment]
I tried recreating this in both singleplayer and multiplayer by placing a crate and putting a portion or all of the crafting materials inside the crate, but the crafting list correctly showed the item as non-craftable. Are you able to reproduce the bug?

**Comment by hannes213** [comment]
It seems to me that all the time when I have stuff in my inventory like 1 flashlight and i go to crafting to scrap it it first shows that I have got 2 flashlights and it is like that with everything

**Comment by hannes213** [comment]
Or is it supposed to show that i have double the amount i have ?

**Comment by hannes213** [comment]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/132841801/935ae224-ae87-4074-872a-0c781eb09b4e)
Like in that image it shows that i have 2/2 chemicals, but i only have 1 in my inventory

**Comment by SDGNelson** [comment]
Is the second chemical in a crate? I'll see if I can reproduce it.

**Comment by SDGNelson** [comment]
I tested what I think is your example - one chemicals in inventory and one either nearby or in storage - and the Horde Beacon crafting entry only showed one chemicals available. Does that sound correct or was something different in your case?

**Comment by hannes213** [comment]
No I didnt even have it in storage container
I only had 1 chemical in my inv and one on the ground far away from me, so that it wouldnt affect me.

**Comment by SDGNelson** [comment]
I just tried reproducing by chemicals on the ground nearby, but it still only counted the one in my inventory. Closing this for the meantime sorry because I can't reproduce it, but if you run into it again and have some steps for how I can reproduce the bug please feel free to comment and I will re-open.

**Comment by Dyna1One** [comment]
Having this same exact issue and found a lot of very old posts about it, I've been running into this issue as well on my server (Russia map) I've tried recreating it in singleplayer but haven't found success yet. Been trying to find the cause for hours now and it got me scratching my head, are you using any mods Hannes213?

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/134254563/35a359e9-85d1-4460-a9c1-032fa5cbe555

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/134254563/5e4c35f9-a3e1-4c41-826f-33c93ad43d3e

**Comment by Dyna1One** [comment]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/134254563/391c6949-cf1e-4532-a61f-0ef5fc176f56)
This is the error message that pops up on my server,

**Comment by SDGNelson** [comment]
That is very interesting and weird that it's showing the wrong number of items! Thanks for the videos and error. I'll read through the related code and see if anything comes to mind.

**Comment by Dyna1One** [comment]
Sure thing! Let me know if I can provide anything, I could also test it on a different client from my laptop to see if it happens for other players, I'm not sure if it's a character related issue, server or map (Though it's interesting that we're both on Russia)

**Comment by SDGNelson** [comment]
So bizarre, I even tried re-creating the same loadout as yours in multiplayer on Russia:
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/12567645/251f58cf-11b0-4b55-b3cb-383fd300546f)
but it showed 1/1 Bulldog rather than 2/1

The inventory/crafting code is pretty old and bad so it's certainly possible there's something going wrong. Maybe a good incentive to make rewriting it higher priority. Just in case though: do you have any plugins on that server, or is it only vanilla?

**Comment by Dyna1One** [comment]
So I've been using the following workshop mods:
- Better Backpacks (fixed) (This is based on Toxicity's Backpacks)
- Alicepack+
- Craft Alicepack (Very unlikely that this would cause issues)
- More Stacking Mod (Updated) - The most likely cause due to the amount of items it adds, as far as I know no dupe IDs etc. though.
- Extended Storage
- Extended Wall Locker

No other mods or plugins (there's some weird plugin selling websites or lack of documentation/descriptions which I'm really not about haha

I've ran a server for a few days on Arid (pure vanilla otherwise, no mods/plugins) which had the same issue after a while of playing..
Also would like to add that I've been running the game with these mods when I tested it in singleplayer myself, with the same items in my inventory and it was working just fine which is... Bizarre haha

**Comment by hannes213** [comment]
I think I should specify that I was playing vanilla singleplayer so I don't think it's due to any plugins or mods ect.

**Comment by hannes213** [comment]
But I dont know, maybe it is due to them and I have something different.

**Comment by Dyna1One** [comment]
Yeah it's a strange one, I was kind of hoping it was just a mod issue but it seems to be happening on vanilla for some people as well (most posts I've found are a lot older however)

**Comment by SDGNelson** [comment]
In the meantime I'll put a potential fix for the server exception, and hopefully the underlying cause comes to light. :confused:

**Comment by GazziFX** [comment]
I think some of your items has amount not equal to 1, I had crafting issues with stacked items and patched them with plugin

**Comment by Alniroza** [comment]
I got the same now, but got no real information to how to replicat, it just "happened".
I have a Almost vanilla Arid server, with just some plugins and a kill feed mood.

Any insights how to fix it??
Thanks!!

**Comment by Alniroza** [comment]
I will give some insights that maybe could be useful:

First, in Arid, the only object i have seen this problems is with Gummed Tape, and almost all recipes related to gummed tape.
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/22087391/d90c8bf5-8e71-4c95-a0c3-c104524a7cf5)

Also, i can see that there only some Gummad Tapes that become duplicated, not everyone, its like a property of the item itself.
I tested this droping and picking up GT. I had 3, and seeing 4 on crafting. 2 Of them counted as one on the crafting recipe, but one in specific was bugged, and counted as 2.

Also, when i trade with merchants, some items that i sell count as 2, this is really strange. For example, i had a Motor, and the trader let me sell that motor twice, and i got the EXP for both of the trades. This also happens with some other items, but i dont know if with everyone.

In the next example, i had 5 wrenches that counted as 9, and 1 photo, that counted as 2.
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/22087391/02fbf49e-692b-4df9-8cbd-a40600bfab6a)

This only let me guess that the duplication count is bound to the object in some way or another. The only property i changed that could been related is  this:  "Resource_Drops_Multiplier": 1.0. I once changed it to 1.5, so maybe that affected in someway, and bounded some items to count as twice?? Anyway, even when i set this back to 1.0 the problem persisted.

I hope that this help you with something.

**Comment by Alniroza** [comment]
Aditional info:
I have trying changing a lot of thing on the server, and the bugged persisted. Also explored the .dat files of the objects on arid, everything seem normal.

Also, i tried reinstaling and nothing changed. This could mean the problem is on server by any way?

Lastly, my server console show no errors.

**Comment by SDGNelson** [comment]
@Alniroza from your description it sounds like the item has an amount/quantity other than `1`, but the client is only showing `1`. I don't think it's caused by any of your config. Maybe a server-to-client item syncing bug. I'll make a note to double-check that the client should show the amount label if not `1` 🤔

**Comment by Alniroza** [comment]
I would really like to help with this, dont know if you could need any particular information.
Or even, you can enter my server to test it yourself, is currently on the server list as "LATAM | DESIERTO DE ATACAMA ... ".
I know this aint happening only to myself, but i dont know if the players are as aware as me, i would have to ask.

Its becoming quite anoying and i really dont want to have to restart the whole server even to test if it got fixed.

If not, there is anything i can try myself to fix it or to give more insights??

**Comment by Alniroza** [comment]
Looks like the problem was Quality_Multiplier. I had it on 1.2, and when i switched back to 1.0 or less, it got fixed.

Previous items kept being bugged, but not new picked up ones.
